### ISSUE TITLE
Listing all tables can make DBAL mad

### ISSUE THREAD

**Issue by CarsonF** [bolt/bolt#2769]
DBAL's Schema Manager tries to create objects for all tables in a database, but fails if it doesn't recognize a type ([here](https://github.com/doctrine/dbal/blob/v2.5.1/lib/Doctrine/DBAL/Platforms/AbstractPlatform.php#L423)).

Bolt should *only* care about tables its own tables (with the prefix specified).

Unfortunately the only way I can find to get around this is by overridding the platform classes.
`getListTablesSQL` needs another where condition for our prefix.

[SqlitePlatform](https://github.com/doctrine/dbal/blob/v2.5.1/lib/Doctrine/DBAL/Platforms/SqlitePlatform.php#L435)
[MySqlPlatform]
(https://github.com/doctrine/dbal/blob/v2.5.1/lib/Doctrine/DBAL/Platforms/MySqlPlatform.php#L349)
[PostgreSqlPlatform](https://github.com/doctrine/dbal/blob/v2.5.1/lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php#L259)

**Comment by GawainLynch** [comment]
Thar be dragooons!

**Comment by GawainLynch** [comment]
This is corner case, and I am not sure this should be on the cards for 2.1. The storage and DB layers are planned to be attacked in 2.2 anyway.  

@bobdenotter?

**Comment by CarsonF** [comment]
Yeah this doesn't need to happen for 2.1

**Comment by bobdenotter** [comment]
Ok, let's bump to 2.2! :-)

**Comment by GawainLynch** [comment]
Kicking to 2.3

**Comment by GawainLynch** [comment]
The platform class gets set in `\Doctrine\DBAL\DriverManager` which is a final class that is called statically, and the hash map of platform drivers is private.

Wondering if we shouldn't allow our config to:

```yaml
mappings:
    mysql:
        - ['bit', 'integer']
```

And then in postConnect()
```php
$db->getDatabasePlatform()->registerDoctrineTypeMapping($dbType, $doctrineType);
```

**Comment by GawainLynch** [comment]
Fixed in #3949
